apps/desktop-wallet/src/features/tokenPrices/hooks.tsx

```diff
@@ -0,0 +1,99 @@
+*/
+
+import { AddressHash, Asset, calculateAmountWorth, client, ONE_MINUTE_MS, TOKENS_QUERY_LIMIT } from '@alephium/shared'
+import { ALPH } from '@alephium/token-list'
+import { explorer } from '@alephium/web3'
+import { useQueries } from '@tanstack/react-query'
+import { chunk, orderBy } from 'lodash'
+import { useMemo } from 'react'
+
+import { useAppSelector } from '@/hooks/redux'
+import {
+  makeSelectAddressesKnownFungibleTokens,
+  makeSelectAddressesListedFungibleTokenSymbols
+} from '@/storage/addresses/addressesSelectors'
+
+export const useAddressTokensPrices = () => {
+  const currency = useAppSelector((s) => s.settings.fiatCurrency).toLowerCase()
+  const addressTokensSymbols = useAppSelector(useMemo(makeSelectAddressesListedFungibleTokenSymbols, [])) // TODO: To be replaced when tokens are fetched with Tanstack
+  const addressTokensSymbolsWithPrice = addressTokensSymbols.filter((symbol) => symbol in explorer.TokensWithPrice)
+
+  const { data, isPending } = useQueries({
+    queries: chunk(addressTokensSymbolsWithPrice, TOKENS_QUERY_LIMIT).map((symbols) => ({
+      queryKey: ['tokenPrices', symbols, { currency }],
+      queryFn: async () =>
+        (await client.explorer.market.postMarketPrices({ currency }, symbols)).map((price, i) => ({
+          price,
+          symbol: symbols[i]
+        })),
+      refetchInterval: ONE_MINUTE_MS
+    })),
+    combine: (results) => ({
+      data: results.flatMap(({ data }) => data).filter((price) => !!price),
+      isPending: results.some(({ isPending }) => isPending)
+    })
+  })
+
+  return { data, isPending }
+}
+
+export const useAlphPrice = () => {
+  const { data: tokenPrices } = useAddressTokensPrices()
+
+  return tokenPrices.find((token) => token.symbol === ALPH.symbol)?.price
+}
+
+export const useSortTokensByWorth = (tokens: Asset[]) => {
+  const { data: tokenPrices } = useAddressTokensPrices()
+
+  const tokensWithWorth = tokens.map((token) => {
+    const tokenPrice = tokenPrices.find((t) => t.symbol === token.symbol)
+
+    return {
+      ...token,
+      worth: tokenPrice ? calculateAmountWorth(token.balance, tokenPrice.price, token.decimals) : undefined
+    }
+  })
+
+  return orderBy(
+    tokensWithWorth,
+    [
+      (a) => (a.verified ? 0 : 1),
+      (a) => a.worth ?? -1,
+      (a) => a.verified === undefined,
+      (a) => a.name?.toLowerCase(),
+      'id'
+    ],
+    ['asc', 'desc', 'asc', 'asc', 'asc']
+  )
+}
+
+export const useAddressTokensWorth = (addressHashes?: AddressHash[] | AddressHash) => {
+  const { data: tokenPrices } = useAddressTokensPrices()
+  const selectAddressesKnownFungibleTokens = useMemo(makeSelectAddressesKnownFungibleTokens, [])
+  const tokens = useAppSelector((s) => selectAddressesKnownFungibleTokens(s, addressHashes))
+
+  return tokenPrices.reduce((totalWorth, { symbol, price }) => {
+    const verifiedFungibleToken = tokens.find((t) => t.symbol === symbol)
```

Comment:
I really like the use of type guards in my branch, which really helped make things clearer and stricter.
Here for instance, you could use a type guard like `isListedToken(t)`.

Also, let's stop using "verified" and use "listed" instead, WDYT?

Reply:
Updated!